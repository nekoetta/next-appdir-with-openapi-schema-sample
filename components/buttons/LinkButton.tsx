'use client'
import { Button } from 'flowbite-react'
import Link from 'next/link'
import { ReactNode } from 'react'

export const LinkButton = ({
  href,
  children,
}: {
  href: string
  children: ReactNode
}) => {
  return (
    <Link href={href} className="block w-fit">
      <Button>{children}</Button>
    </Link>
  )
}
