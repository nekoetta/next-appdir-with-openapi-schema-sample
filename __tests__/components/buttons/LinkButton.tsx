import { render, screen } from '@testing-library/react'
import { LinkButton } from '@/components/buttons/LinkButton'
import '@testing-library/jest-dom'

describe('LinkButton', () => {
  it('renders a link button', () => {
    const { container } = render(
      <LinkButton href="test_link">
        <div>child</div>
      </LinkButton>
    )
    expect(container.getElementsByTagName('a')[0].href).toContain('test_link')
    expect(screen.getByText('child')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
