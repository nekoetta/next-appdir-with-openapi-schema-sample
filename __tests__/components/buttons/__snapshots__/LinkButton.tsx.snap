// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders a link button 1`] = `
<div>
  <a
    class="block w-fit"
    href="test_link"
  >
    <button
      class="text-white bg-blue-700 border border-transparent hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 disabled:hover:bg-blue-700 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 dark:disabled:hover:bg-blue-600 focus:!ring-2 group flex h-min items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg"
      type="button"
    >
      <span
        class="flex items-center rounded-md text-sm px-4 py-2"
      >
        <div>
          child
        </div>
      </span>
    </button>
  </a>
</div>
`;
